test/cpp/end2end/xds_end2end_test.cc

```diff
@@ -7255,12 +7332,54 @@ TEST_P(CdsTest, MultipleBadResources) {
   EXPECT_EQ(response_state.state, AdsServiceImpl::ResponseState::NACKED);
   EXPECT_THAT(
       response_state.error_message,
-      ::testing::ContainsRegex(absl::StrCat(kDefaultClusterName,
+      ::testing::ContainsRegex(absl::StrCat(kClusterName2,
                                             ": validation error.*"
                                             "DiscoveryType is not valid.*",
-                                            kClusterName2,
+                                            kClusterName3,
                                             ": validation error.*"
                                             "DiscoveryType is not valid")));
+  // RPCs for default cluster should succeed.
+  std::vector<std::pair<std::string, std::string>> metadata_default_cluster = {
+      {"cluster", kDefaultClusterName},
+  };
+  CheckRpcSendOk(
+      1, RpcOptions().set_metadata(std::move(metadata_default_cluster)));
+  // RPCs for cluster 2 should fail.
+  std::vector<std::pair<std::string, std::string>> metadata_cluster_2 = {
+      {"cluster", kClusterName2},
+  };
+  CheckRpcSendFailure(CheckRpcSendFailureOptions().set_rpc_options(
+      RpcOptions().set_metadata(std::move(metadata_default_cluster))));
```

Comment:
metadata_default_cluster was moved a second time

Reply:
Thanks for catching this!  That's actually a bug in the test; it was supposed to use the second set of metadata, not reuse the first set.

Fixed.